### PR TITLE
Add LoggerListener

### LINKED ISSUE #5148: Add LoggerListener

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
Fixes #5148

### CODE REVIEW

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

File: src/SLCore.UnitTests/Listener/LoggerListenerTests.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.SLCore.Core;
+using SonarLint.VisualStudio.SLCore.Listener;
+using SonarLint.VisualStudio.TestInfrastructure;
+
+namespace SonarLint.VisualStudio.SLCore.UnitTests.Listener
+{
+    [TestClass]
+    public partial class LoggerListenerTests
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SLCore.UnitTests/Listener/LoggerListenerTests.cs:29]: Why partial class? One is for listener, one is for serialization. 2 different things

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]

File: src/SLCore.UnitTests/Listener/LoggerListenerTests.cs (hunk 1 above)
Comment (ugras-ergun-sonarsource) [reply src/SLCore.UnitTests/Listener/LoggerListenerTests.cs:29]: They are related to logging So I wanted to group them together.

Comment (ugras-ergun-sonarsource) [reply src/SLCore.UnitTests/Listener/LoggerListenerTests.cs:29]: But seperate enough so partial class

Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.UnitTests/Listener/LoggerListenerTests.cs:29]: I disagree. We wouldn't have separate classes for SQ client tests then. The 2 things are different. Partial classes are confusing

Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.UnitTests/Listener/LoggerListenerTests.cs:29]: For grouping - use folder/namespace

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**